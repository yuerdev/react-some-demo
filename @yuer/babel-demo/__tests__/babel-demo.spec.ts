const { transformFileSync } = require('@babel/core');
import { dirname } from "path";
import { fileURLToPath } from "url";
import insertParametersPlugin from '..';
const path = require('path');


const __dirname = dirname(fileURLToPath(import.meta.url));
describe("babel-demo", () => {
    it("babel-demo", () => {
        const { code } = transformFileSync(path.join(__dirname, './sourceCode.js'), {
            root: __dirname,
            parserOpts: {
                plugins: ['jsx']
            },
            plugins: [insertParametersPlugin],
        });

        expect(code).toMatchSnapshot();
    })
})