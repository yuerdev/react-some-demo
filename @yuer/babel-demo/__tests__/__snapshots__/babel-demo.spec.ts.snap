// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-demo babel-demo 1`] = `
"console.log(\\"unkown filename: (2, 0)\\")
console.log(1);

function func() {
  console.log(\\"unkown filename: (5, 4)\\")
  console.info(2);
}

export default class Class {
  say() {
    console.log(\\"unkown filename: (10, 8)\\")
    console.debug(3);
  }

  render() {
    return <div>{[console.log(\\"unkown filename: (13, 21)\\"), console.error(4)]}</div>;
  }

}"
`;
